### PR TITLE
[Move] Rototiller implementation

### PR DESCRIPTION
**Pull request by Raidette**
## What are the changes?
Added Rototiller, which gives +1 Atk. and Sp.Atk to every grounded plant pokémon on the battlefield.

### Screenshots/Videos
Singles Example
https://github.com/pagefaultgames/pokerogue/assets/73134872/70e7debe-5a17-42b1-a5af-70786b55c608
Doubles Examples
https://github.com/pagefaultgames/pokerogue/assets/73134872/ea9c5bdc-794c-4416-a166-35351a2a1392

## How to test the changes?
Play a non-grass, grounded grass and ungrounded grass type pokémon, and make any of them use Rototiller.

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/data/move.ts [hunk 1]
```diff
@@ -6683,7 +6683,7 @@ export function initMoves() {
       .condition((user, target, move) => user.battleData.berriesEaten.length > 0),
     new StatusMove(Moves.ROTOTILLER, Type.GROUND, -1, 10, 100, 0, 6)
       .target(MoveTarget.ALL)
-      .unimplemented(),
+      .attr(StatChangeAttr, [BattleStat.ATK, BattleStat.SPATK], 1, false, (user, target, move) => target.isGrounded() && target.isOfType(Type.GRASS)),
```
Comment (Tempo-anon) [inline src/data/move.ts:6686]: Does this account for Pokemon that are using fly/dig/dive/shadowforce/etc?

Also, does this move fail if there are no eligible grass Pokemon?

Comment (Raidette) [reply src/data/move.ts:6686]: Currently :
- It does not work for pokémon in the semi-invincible turn of a two-turn move, as expected
- It does not fail (move animation plays but nobody gets boosted) if there are no eligible grass pokémon, will fix that next commit

**Comment by Raidette** [comment]
Attack now fails if no eligible grass type is fielded, it still triggers the animation if it is in a semi-invulnerability state, but the pokémon dodges the attack.

**Comment by Tempo-anon** [comment]
Can you add the video examples of it failing as well as what happens when there are pokemon flying/diving/etc?

**Comment by Raidette** [comment]
## Failing with no eligible grass type (Carnivine has levitation) 
https://github.com/pagefaultgames/pokerogue/assets/73134872/df62e1b7-d668-4132-b9d4-58daf4bbb9aa

## Used with eligible grass types but in semi-invulnerability
https://github.com/pagefaultgames/pokerogue/assets/73134872/4807fce6-3c69-4028-adf7-066050aab5f1

**Review by Tempo-anon — APPROVED** [review]

**Pull request merged**